src/compiler/checker.ts

```diff
@@ -21871,6 +21871,10 @@ namespace ts {
                 // and give a better error message when the host function mentions `arguments`
                 // but the tag doesn't have an array type
                 if (decl) {
+                    const i = getJSDocTags(decl).filter(isJSDocParameterTag).indexOf(node);

```

Comment:
Maybe `getParameterSymbolFromJSDoc` should work for binding patterns too?

Reply:
There's no symbol on a binding pattern, though, so without adding something in the binder there's no benefit.